src/widgets/lv_img.c

```diff
@@ -410,7 +414,28 @@ static lv_res_t get_metadata(lv_img_t * img)
         header.cf = LV_IMG_CF_ALPHA_1BIT;
     }
     else {
-        lv_img_decoder_get_info(&img->src, &header);
+        /*Query the image cache now, since it's very likely we'll draw the image later on,
+          so don't waste opening a decoder and closing it if it'll be reused */
+        lv_img_dec_dsc_in_t dsc = {0};
+        dsc.src = &img->src;
+        /*Don't waste a cached entry for a different color, use the color that'll be drawn*/
+        dsc.color = lv_obj_get_style_img_recolor_filtered(obj, LV_PART_MAIN);
+        dsc.size_hint.x = lv_obj_get_style_width(obj, LV_PART_MAIN);
+        dsc.size_hint.y = lv_obj_get_style_height(obj, LV_PART_MAIN);
+
+        lv_img_cache_entry_t * entry = lv_img_cache_open(&dsc, img->dec_ctx);
```

Comment:
This the main differences in our opinions. Now I also understand (at least I hope :slightly_smiling_face: ) the whole flow, so let's decide this topic: I think a full open is very wasteful as it will render the image too. An `lv_img_decoder_get_info` seems more appropriate.  Setting an image source is a very rare event compared to rendering and image over and over. So I think caching shouldn't be used on this level at all to keep things simpler. 

I see that getting info from lottie takes more time than getting info from a PNG, but rendering a lottie animation is also very performance hungry. "Accidentally" rendering the first frame of some lottie anims compared to getting the info without caching seems much larger problem.

Reply:
Ok, just to explain better here:
1. When calling `decoder_open` you have currently 2 way to *open* a picture, either decode it or extract metadata. 
2. Right now, `lv_img_cache_open` by default, find a free decoder entry and decode the picture (slower), while `lv_img_cache_query` tries to find an existing decoder entry to capture the metadata from it and if not found create a temporary and only extract metadata (faster).

So, if we act like you suggest, it means that I'll replace the `lv_img_cache_open` by `...query` instead. So for most usage, it means a "dual" decoding of the picture (the first one is partial to only extract metadata and the second one is complete, with decoding). This is bad for bmp, sjpg, rgbdec, rlottie and ffmpeg decoders, but better for png and gif (only), since the former don't decode until calling `read_line`.

It's not a good idea to skip caching here, since if a picture was previously decoded, it does not make sense to re-decode it instead of using the image's cache metadata.